ddl/callback_test.go

```diff
@@ -63,11 +65,13 @@ func (tc *TestDDLCallback) OnJobRunBefore(job *model.Job) {
 
 func (tc *TestDDLCallback) OnJobUpdated(job *model.Job) {
 	log.Infof("on job updated, job %v", job)
-	if tc.OnJobUpdatedExported != nil {
+	tp := atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&tc.OnJobUpdatedExported)))
```

Comment:
Maybe not cause by this.
```go
Previous write at 0x00c00376d398 by goroutine 257:
  github.com/pingcap/tidb/ddl.(*testForeighKeySuite).TestForeignKey()
      /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/foreign_key_test.go:215 +0x159e
```
look this.
https://github.com/pingcap/tidb/blob/master/ddl/foreign_key_test.go#L180
maybe need a new variable to fix data race, like this:
```go
tc = &TestDDLCallback{}
tc.onJobUpdated = func(job *model.Job) {
...
}
d.SetHook(tc)
```
https://github.com/pingcap/tidb/blob/master/ddl/foreign_key_test.go#L215 this can be remove. 
and use `defer` to set origin hook back before https://github.com/pingcap/tidb/blob/master/ddl/foreign_key_test.go#L157
```go
originalHook := d.GetHook()
defer d.SetHook(originalHook)
```

Reply:
Thanks for your timely help.
I will check it ASAP.